Code/Framework/AzToolsFramework/Tests/Prefab/PrefabUndoAddEntityTests.cpp

```diff
@@ -0,0 +1,335 @@
+        PrefabUndoAddEntityAsOverride undoAddLeftWheelEntityNode = createUndoAddLeftWheelEntityNodeResult.TakeValue();
+
+        size_t expectedLeftWheelInstanceEntityCount = 0;
+        size_t expectedRightWheelInstanceEntityCount = 0;
+
+        // Adding the left wheel entity under the left wheel instance by redoing with our undo node and doing template propagation.
+        undoAddLeftWheelEntityNode.Redo();
+        PropagateAllTemplateChanges();
+
+        auto findAxisInstanceResult = focusedCarInstance.FindNestedInstance(axisInstanceAlias);
+        ASSERT_TRUE(findAxisInstanceResult.has_value());
+        Instance& axisInstance = findAxisInstanceResult->get();
+
+        auto findRightWheelInstanceResult = axisInstance.FindNestedInstance(rightWheelInstanceAlias);
+        ASSERT_TRUE(findRightWheelInstanceResult.has_value());
+        Instance& rightWheelInstance = findRightWheelInstanceResult->get();
+
+        // We should see the left wheel entity be added under left wheel instance.
+        ValidateNewEntityUnderInstance(
+            leftWheelInstance, leftWheelEntityAlias, leftWheelEntityName, ++expectedLeftWheelInstanceEntityCount);
+
+        // The focused instance is the car instance, which is the ancestor instance of
+        // the owning instance of our new wheel entity (left wheel instance).
+        // After propagation, we should see the new wheel entity has been added under 
+        // left wheel instance, not right wheel instance.
+        ValidateNewEntityNotUnderInstance(
+            rightWheelInstance, leftWheelEntityAlias, expectedRightWheelInstanceEntityCount);
+
+        // Create a tire entity and add it under our left wheel entity.
+        const AZStd::string tireEntityName = "Tire";
+        const EntityAlias tireEntityAlias = CreateEntity(tireEntityName, leftWheelInstance, leftWheelEntityAlias);
+        ASSERT_FALSE(tireEntityAlias.empty());
+
+        // Create undo/redo node for adding the tire entity under the left wheel entity.
+        auto createUndoAddTireEntityNodeResult = CreatePrefabUndoAddEntityAsOverrideNode(
+            leftWheelEntityAlias, tireEntityAlias, leftWheelInstance, focusedCarInstance, "Undo Adding Tire Entity");
+        ASSERT_TRUE(createUndoAddTireEntityNodeResult.IsSuccess());
+        PrefabUndoAddEntityAsOverride undoAddTireEntityNode = createUndoAddTireEntityNodeResult.TakeValue();
+
+        // Adding the tire entity under the left wheel entity by redoing with our undo node and doing template propagation.
+        undoAddTireEntityNode.Redo();
+        PropagateAllTemplateChanges();
+
+        ValidateNewEntityUnderParentEntity(leftWheelInstance, leftWheelEntityAlias, leftWheelEntityName,
+            tireEntityAlias, tireEntityName, ++expectedLeftWheelInstanceEntityCount);
+
+        ValidateNewEntityNotUnderInstance(rightWheelInstance, tireEntityAlias, expectedRightWheelInstanceEntityCount);
+
+        // Create a frame entity and add it under our left wheel entity.
```

Comment:
Adding wheel and tire entities already proves that you can add an entity as an override under an already overridden entity. The addition of frame and nail entities seems repetitive and not necessary. If you still feel like adding them, consider adding a for loop for each addition and validation rather than writing big test cases like this.